### PR TITLE
Add Makefile targets for easily running and checking yapf auto-formatting

### PR DESCRIPTION
**Pull request by nicksay**
Progress on #21

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,6 +41,13 @@ spitfire/runtime/_baked.so spitfire/runtime/_template.so spitfire/runtime/_udn.s
 	$(PIP) install --user --editable .
 
 
+fix:
+	$(YAPF) --in-place --recursive --verify spitfire scripts || (echo; echo 'Files were auto-formatted. Please double-check the changes made.')
```
Comment (awbraunstein) [inline Makefile]: I think echo '\nFiles...' is cleaner than echo; echo 'Files...

File: Makefile [hunk 2]
```diff
@@ -3,13 +3,21 @@
...
 .PHONY: tests unit_tests no_whitespace_tests whitespace_tests test_function_registry test_opt
+# Clean up
 .PHONY: clean clean_build clean_tests
+# Format code
+.PHONY: fix lint
+
 
 PYTHON ?= python
 PIP ?= pip
+YAPF ?= yapf
```
Comment (awbraunstein) [inline Makefile]: Should we have some sort of setup script that ensures that yapf is installed?

File: Makefile (hunk 1 above)
Comment (nicksay) [inline Makefile]: Unfortunately, make runs the `echo` command, not the bash built-in, and doesn't support the `-e` flag.  I adjusted the output to make it a little clearer.

File: Makefile (hunk 2 above)
Comment (nicksay) [inline Makefile]: Yeah, we'll need to deal with development-time deps.

File: Makefile (hunk 1 above)
Comment (awbraunstein) [inline Makefile]: Interesting. I didn't know that. Well you learn something new everyday.

**Comment by awbraunstein** [comment]
lgtm. :palm_tree:

**Pull request merged**